### ISSUE TITLE
Merge strategy for PRs

### ISSUE THREAD

**Issue by xtough** [SAP/styleguides#184]
Shall we set the merge strategy to "Squash+Merge" by default on this repository?

**Comment by HrFlorianHoffmann** [comment]
Ooh, yes, please!

**Comment by xtough** [comment]
@larshp @filak-sap any opinions?

**Comment by HrFlorianHoffmann** [comment]
Looks like I cannot select the default. What I could do however is disable the plain merge and rebase actions.

**Comment by xtough** [comment]
For the sake of a concise commit history, I could live with disabling plain merge and rebase options.
I could imagine that others (like @filak-sap ) may have a different opinion, though.

**Comment by filak-sap** [comment]
- Merge commits make sense when you are in a big project and you want to have a history of Added features and contributors can make good commits.
- - But even simple Merge should be done with "Rebase the branch on the main branch and then merge with Fast Forward", so nobody has to deal with interleaved commits in "git log" (or in UIs).
- Rebasing is for those who wants to have a repo without Merge commits - mainly repos with not so much traffic.
- Squash is good when you have a contributors who cannot use git efficiently and cannot do `git commit --amend` and `git rebase -i` or when you have advanced contributors who split their changes into several commits to help reviewers to understand the changes.

Bottom line, I turn of "Merge commits" in my repos because simple merging leads to hell of interleaved commits - unless you know to use "git log --topo-order".

**Comment by filak-sap** [comment]
Using Squash without considering the PR commits may upset contributors who prepare commits carefully.

**Comment by xtough** [comment]
Looks like an impasse. 3 👍 for removing merge&rebase so that only squashing is possible vs. @filak-sap 's valid objections about squashing as the only option. There seems to be a consent about disabling plain merge, though (leaving squash and rebase options). Shall we go for that?

**Comment by xtough** [comment]
@HrFlorianHoffmann : since @filak-sap is ok with removing at least the plain merge option and I lack permissions to do that, would you please configure that?

**Comment by HrFlorianHoffmann** [comment]
Done. Squashing and rebasing remain active.
